xformers/triton/k_layer_norm.py

```diff
@@ -246,38 +323,44 @@ def backward(ctx, dy):
 
         # flatten the batch dimension, if any.
         # We're interested in 'samples' x norm_dimension
-        x_arg = x.reshape(-1, x.shape[-1])
-        M, N = x_arg.shape
+        y = y.reshape(-1, y.size(-1))
+        M, N = y.size()
 
         # allocate output
         locks = torch.zeros(2 * GROUP_SIZE_M, dtype=torch.int32, device="cuda")
-        t_args = {"dtype" : x.dtype, "device" : x.device}
-        _dw = torch.empty((GROUP_SIZE_M, weight.shape[0]), **t_args)
-        _db = torch.empty((GROUP_SIZE_M, weight.shape[0]), **t_args)
-        dw = torch.empty((weight.shape[0],), **t_args)
-        db = torch.empty((weight.shape[0],), **t_args)
+        t_args = {"dtype": y.dtype, "device": y.device}
+        _dw = torch.empty((GROUP_SIZE_M, y.size(-1)), **t_args)
+        _db = torch.empty((GROUP_SIZE_M, y.size(-1)), **t_args)
+        dw = torch.empty((y.size(-1),), **t_args)
+        db = torch.empty((y.size(-1),), **t_args)
         dy = dy.contiguous()
         dx = torch.empty_like(dy)
 
         # Check the tensor shapes and layouts
         # we suppose in the kernel that they have the same size and are contiguous
-        assert dx.numel() == x.numel(), \
+        assert dx.numel() == y.numel(), \
             "Something is wrong in the backward graph, possibly because of an inplace operation after the layernorm"
 
         # enqueue kernel using forward pass heuristics
         # also compute partial sums for DW and DB
 
         # fmt: off
+        meta = {"BLOCK_SIZE_N": ctx.BLOCK_SIZE_N,
```

Comment:
you probably saw that, but this is the same as passing it directly down when calling the kernel

Reply:
I could integrate it into the two calls below. It just felt cleaner to not repeat shared arguments.